File: Data Structures & Algorithms/is-subsequence/submission-0.java
class Solution {
    public boolean isSubsequence(String s, String t) {
        if (t.length() == 0){
            return false; 
        }

        if (s.length() == 0){
            return true;
        }

        int left = 0;
        int right = 0;

        while (left < s.length() && right < t.length()){

            // check if we found the chars matched if so increment the left 
            if (s.charAt(left) == t.charAt(right)){
                left++;
            }

            // alwas increment the right
            right++;
        }
        return left == s.length(); // if s is a subsequece of t the left value should be same as s.length
    }
}

File: Data Structures & Algorithms/squares-of-a-sorted-array/submission-0.py
class Solution:
    def sortedSquares(self, nums: List[int]) -> List[int]:
        n = len(nums)
        
        for i in range(n):
            nums[i] = nums[i] ** 2
        
        left = 0
        right = n-1
        na = n-1

        newArray = [0] * n
        while left <= right:
            if nums[right] > nums[left]:
                newArray[na] = nums[right]
                right -= 1
            else:
                newArray[na] = nums[left]
                left += 1
            na -= 1

        return newArray